airflow/task/task_runner/base_task_runner.py

```diff
@@ -105,7 +105,8 @@ def deserialize_run_error(self) -> Optional[Union[str, Exception]]:
         """Return task runtime error if its written to provided error file."""
         return load_error_file(self._error_file)
 
-    def _read_task_logs(self, stream):
+    def _stream_reader(self, stream):
+        """Reading from the stream and log into task logs"""
         while True:
             line = stream.readline()
             if isinstance(line, bytes):
```

Comment:
I was tempting to rewrite this as:
```
for line in iter(stream, ""):
    ...
```

I wonder if both binary and string would get thrown into the stream? so we need to handle both `bytes` and `str`?

Reply:
The only place `_read_task_logs()` is called, `stream` is the stdout of a `Popen(universal_newlines=True)`, so it can actually never emit bytes anywhere in the code base. And since `_read_task_logs()` is private, any third-party code that relies on it is subject to break anyway.

I would categorise this as *overly defensive* programming in the original implementation and simply remvoe the `if isinstance(line, bytes)` check entirely. It should not have been written in the first place.

Reply:
Thanks for the suggestion! Now I think about it deeper, it might be originated from the period when we need to maintain both python2 and python3. Regardless, as we don't support python2 anymore, we should remove it if there is no other reason to keep it.